t-SNE API.ipynb

```diff
@@ -0,0 +1,338 @@
upKV57zYw4834d/KeAR0vw/QD8ZwFO/vnnIk2btsAwHjyX8R1F5/o59bd0\nWwiaDxcPwY9BWFcNJqhtc34PXhsvRblGo3kExDqNpqQlbRHJAfyA8uNYi1Iemkgpg83zeXFxCJVS\nbkMpGS+jii61A9pkhBwcoC0cmkfAwYMHqVs3kOvXb2MYpbBaDzJr1iw+/fRTPvjgAwDWrl1L//5v\nmz3+QOnCse+hhOnEs6BKtp8wH3thtZ6nQIF7121p27YtwcHrGDVqNDt27OT6jYuIreVx5uwC0Rex\nnJ9Fjdp1adu2baL9IyMjkTIGPNrHP+HRBrDidNYmJGQJq1evpnnz5im4O3FUrVqVQkWKcWLFO/DC\ncvDNBQXrgWdWfGU4/549nWiGU41G8whwkvItkoRhawmQUvZO4vxTiRxbiNp7znBoC4cmzXn99Te4\nft0dw5gIDMAwxgJt+PDDDwkODqZFixY0btyEmJicQHegF6qwGSirRsL8N0dR7/zzKOvHcgzjNK+9\nlpiPVRwNGjRg1aoVhIWFsnfPbto1r4LftSnks6xl0PvvsGb1Smy2xHXwrFmzIiwWcCRwHHccM9dS\nDCGsLv4iKUcIQZVKFeDCPhhbACaWg9H54cIBrodd5dSpUw88tkajSSWOB2yaO2gLhyZNCQ8PZ/36\ndSgLYGz5c4HaFvmNRo2eRv0beqC2L2O3CvICX6PskrtQUV8lUc7Ym1C6shNYgs1m59NPx9CsWbNk\nr6tcuXLMnz8v2fJZs2alTeu2/Lp8KIatFGQKBOMEXO2J2ubxREqDEiVKJDXUPZFSsnLVGsjRDzIV\ngtv/QP4Xwb8ztgPlmDNnDiNGjHjg8TUaTSrQ1WJTjVY4NGmKKgEP8TOAAtiQUqC2RnKhfDZcZSxA\nUVSiraaonBq7AUFgYD2mTZvGtm3bkFLSrFmzR5Lu+5tvptC4SXP27mkAwgvkLZSfV0tstkmUKFGe\np566ywKaIiJu3YCAQpCrX7zjwu7HjRs3UjW2RqNJBWmU+Ou/hN5S0aQpAQEBVK5cDYtlDRDtcmat\n+bwYEACcI/6Gp0SFwvqholHeRoWmS7p06UKxYsXo1q0b3bt3fyjKxuHDh9m4cSNXr169p0y2bNnY\nvWsny5Yto0b1slgsVlR47E/UqlWGVat+S1bdl3shhKBeYAOsoTPB6XKvwjcQc/MYDRs2fOCxNRqN\nJr3RCocmzZk48XNsttPYbO8BM4FRwPfm2VOoZFrXgcXAVVRSrZUoJaS6KWcFLgHgdDrp06cP3bp1\nY+bMmURGRj7w2k6cOEGNmnUoVaoU9evXJ1fuPLz77oB7RptYLBZatmzJ9u3bOX/+X9avX88///zD\nxo2/kydPngdeRywjR3yCiNyH9WANODcWTryJ9WhLataqQ6tWrVI9vkajeUBinUZT0pJwGv2vkWYK\nhxBikBBiixDilhAi0Z+NQoh8QojlpswFIcQYs4CNq0wDIcQuIUSkEOKIEKJ7IuP0FUKcEELcFkJs\nF0JUS6vr0qScevXqsXPndtq3b4Dd/jtwAJUqvD3KurECleDrMCqfzQRgp9n7NEop2YHVupQ8efLS\np08fpk1byJw5m+jRowc1a9YhPDxZKYTjERMTQ6Onm7L7wEUouxCqHyA61/uMHz+ekSNHJtk/e/bs\nNGjQgOLFi6d47ntRp04dNvy+noY1c5Dp4lCyGj/z9puv3dehVaPRPAJifThS0rQPRzzS0sJhR1XB\nmpzYSVOx+A3lR1ITFZ7QA/jERaYgcdW1KqCSnkwTQjR2kemMqp43FKgE7AVWCSESxlJq0pEKFSrQ\nvHkzYmJuo7ZGeqLSlvdDhZlvR9UugUqVKjF//jwGDBiAp+ceYAYWy2oCA2tw7txZoC0Ox0cYxtvA\nu+zff+iBnCl//fVXToQcxVFyAWRrB16lodBQZJ7XGT/hS2JiUrph+3CoXbs2a1avJDIygsuXzjNm\nzBi8vb3TZS0ajcZER6mkmjRTOKSUw6SUXwD77iHSFBV20EVKuU9KuQpVPKOvECL2p1wfIERK+Z6U\n8h8p5dfAAuAtl3HeAqZKKX8wi928isoK1SsNLkuTCrZs2YLNlhe1hRKLFagMOPD0dDJr1ix27dpF\nhw4dGDNmDJcuXWD//v1cvHiBUqVKYrNlAxqgIl0A8mMY1fnxxzkpXs/hw4exuQeAT4L8HX6NCA8L\n5cqVKw9wlRqN5okkbRJ//adITxttTWCflNL1U30VyiJSBmWpqInyLiSBzAQAIYQdlZnt09iTUkop\nhFiL8jTUPEb4+fmhfDUMlKIRSxg+Pr6cO3cGHx+feH28vLwoU6YMAPv378fhcONuPdmTW7ciUrye\nQoUK4YgMhYgj4OmyLRK+DS9vX/z9/e/dWaPR/LdIg8Rf/zXS02k0J3dXuLvocu5+Mr5CiEyoFJTW\ne8jkRPNY8cILL+BwXEftksWgIlH+wWLZyeuvv3aXsuHKggUL2LhxI8qR9KTLmShstl00bfp0itfz\n7LPPkiNnHqyHOkHYZoi+DOcmYzk3nj6vvkymTPeqn6TRaDSalJIiC4cQYhQw8D4iEiglpTySqlWp\nce65jGTK3O+8Jh0oW7YsEydO5K233sJi+RMhPHA4QqlbtwEfffTRffsOGfoJ+DaBmKtwexJQFZW7\nYwdW622GDh163/6J4e7uzprVK2j7bAdC/qoHqNDU57u+kCynUY1G8x9CJ/5KNSndUhkH3F2OMz7J\nze18AVWly5UcLudi/yZMspAduC6ljBZCXEG9pInJJLR63MVbb70VrzIoQFBQEEFBQUmvXvNA9O/f\nnxYtWvDTTz9x48YNGjRoQLNmze6bvyIqKopDB/dB/rcgS3u4OAquzgbnLQTRBAU9R9myZR9oPeXK\nlePokUNs2bKFS5cuUblyZQoVKvSgl6fRaNKQuXPnMnfu3HjHHiRC7Y
```

Comment:
Bug: `tsne["Band gap"]["label"][outlier_index], tsne["Band gap"]["uid"][max_bg_index])` shouldn't be using the `max_bg_index`.